app/next-client-app/components/scanreports/ButtonsRow.tsx

```diff
@@ -27,11 +29,19 @@ export function ButtonsRow({
       </div>
       <div className="flex gap-2">
         {" "}
-        <Link href={`/scanreports/${scanreportId}/tables/${tableId}/update`}>
-          <Button variant="outline">
-            Edit Table <Pencil className="ml-2 size-4" />
+        {permissions.includes("CanEdit") || permissions.includes("CanAdmin") ? (
```

Comment:
I can see why we need to do this - to find a way to not include the Link in the component based on the permissions. It's frustrating we can't disable the Link
I wonder if a cleaner way might be to extract the Button component itself.

```
const canEdit = permissions.includes("CanEdit") || permissions.includes("CanAdmin");

const EditTableButton = (
  <Button variant="outline" disabled={!canEdit}>
    Edit Table
    <Pencil className="ml-2 size-4" />
  </Button>
);

return canEditOrAdmin ? (
  <Link href={`/scanreports/${scanreportId}/tables/${tableId}/update`}>
    {EditTableButton}
  </Link>
) : (
  EditTableButton
);
```

At least that way we're not duplicating the Button code, still not clean though.

Looking at the Shadcn Button docs, the advised way to use a Link inside a Button is to wrap as following instead:
```
<Button asChild>
  <Link href="/login">Login</Link>
</Button>
```
I'm curious - if the button is disabled in this case, does the link no longer work?

Reply:
Also if we do have to do it the first way, as we use this same pattern 3 times at least already - I think it's worth moving the entire logic out to a `EditButtonLink` component (or something similarly named)